### DISCUSSION TITLE
The server name and version could not be captured by using tls.ConnectionState

### DISCUSSION THREAD

**Discussion by a2htray** [rabbitmq/amqp091-go#211] [Q&A] [answered]
### Describe the bug

Thanks for this great project.

When using `amqp091-go v1.8.1` and `Go 1.19.3`, I found the server name and version could not be captured. The code is simple as follows.

```go
// main.go
conn, err := amqp.Dial("amqp://goadmin:123456@localhost:5672/goapp-vhost")

connState := conn.ConnectionState()
fmt.Println("Server name: ", connState.ServerName)
fmt.Println("Version: ", connState.Version)
```

```bash
$ go run main.go

0
```

### Reproduction steps

1. Create connection with the plain URL
2. call `conn.ConnectionState` to get ConnectionState
3. access `ConnectionState.ServerName` and `ConnectionState.Version`

### Expected behavior

print right server name and version.

**Comment by Zerpet** [comment, marked as answer]
This is working as intended. You are not dialing to an `amqps` address, therefore, `Dial` is not starting a TLS transport. Since it is not using TLS, `ConnectionState` returns a zero-value, as documented in the [function documentation](https://pkg.go.dev/github.com/rabbitmq/amqp091-go#Connection.ConnectionState). See also [tls#ConnectionState](https://pkg.go.dev/crypto/tls#ConnectionState) for the fields you can find and their meaning.

By setting a RabbitMQ server that serves over AMQP+TLS, and running this code:

```go
// main.go
	conn, err := amqp.DialTLS("amqps://127.0.0.1:5671", &tls.Config{
		InsecureSkipVerify: true,
	})
	if err != nil {
		errLog.Fatalf("error dialing: %s", err)
	}
	defer conn.Close()

	connState := conn.ConnectionState()
	infoLog.Printf("Server name: '%s'", connState.ServerName)
	infoLog.Printf("Version: %#x", connState.Version)

// Output
// 2023/07/12 12:58:46 [INFO] Server name: ''
// 2023/07/12 12:58:46 [INFO] Version: 0x304
```

I get the expected results. Note that server name is still empty, because I dialed to an IP address. The number 0x0304 matches the [constant for TLS 1.3](https://pkg.go.dev/crypto/tls#VersionTLS13)

**Comment by a2htray** [comment]
Thanks for your working
